main.py

```diff
@@ -41,9 +45,19 @@ def get_user_input():
     url, maximum_pages = get_user_input()
 
     _stampifier = Stampifier(url, maximum_pages)
+    generated_stamp = ''
 
     try:
         generated_stamp = _stampifier.stampify()
-        print(generated_stamp)
     except Error as err:
         LOGGER.debug(err.message)
+
+    if generated_stamp:
+        # Save the generated_stamp to file
+        try:
+            os.makedirs(STAMP_DIRECTORY, exist_ok=True)
```

Comment:
You can merge this try block with the upper one

Reply:
Makes sense! Updated. Let me know if it requires further changes.